tests/tantivy_test.py

```diff
@@ -418,8 +418,8 @@ def test_create_readers(self):
 
 class TestSearcher(object):
     def test_searcher_repr(self, ram_index, ram_index_numeric_fields):
-        assert repr(ram_index.searcher()) == "Searcher(num_docs=3, num_segments=1)"
-        assert repr(ram_index_numeric_fields.searcher()) == "Searcher(num_docs=2, num_segments=1)"
+        assert repr(ram_index.searcher()) == "Searcher(num_docs=3, num_segments=3)"
```

Comment:
Why did this change and is the reason stable?

Reply:
I did a bisection (not easy, because of the ff changes).  The tantivy commit that introduces the change in this tantivy-py test above is this: https://github.com/quickwit-oss/tantivy/pull/1828/commits/bd5eea98527e74d7f57dd39d62e4ab25ce8a5e5b

Unfortunately it's the big columnar merge, 180 changed files :(

Reply:
We (tantivy-py) may just have to trust that tantivy is doing here with segments is correct. I don't think it's any kind of regression or change on our side.

Reply:
> Number of threads would have been my first idea as well and are also why I asked about stability. To me it looks like for the `RamDirectory`, we always have `num_docs == num_segments`? Maybe @PSeitz has an idea why that might be the case?

Maybe the memory limit is too low, so it commits after every doc

Reply:
In this unit test only 3 tiny documents are added. Is it still possible a memory limit is involved? Is it configurable or do you have a link (to source code presumably) where I can read up more about it?

Reply:
I think that the memory limit involved would be the one passed on [`Index::writer`](https://docs.rs/tantivy/latest/tantivy/struct.Index.html#method.writer) and stored in [`IndexWriter::memory_arena_in_bytes_per_thread`](https://docs.rs/tantivy/latest/src/tantivy/indexer/index_writer.rs.html#60).

Reply:
So from

https://github.com/quickwit-oss/tantivy-py/blob/468166f334b27f11a4a158c7565454bf1b26986b/src/index.rs#L220

I would say the tests use `3_000_000`.

Reply:
I had similar problem in the example in https://github.com/quickwit-oss/tantivy/pull/2130 and bumping the memory to `10_000_000` resolved things there.

Reply:
That was it!  Thx @adamreichold & @PSeitz.